controllers/components/kserve/kserve_support.go

```diff
@@ -0,0 +1,233 @@
+package kserve
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path"
+
+	"github.com/hashicorp/go-multierror"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	componentsv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/components/v1"
+	dsciv1 "github.com/opendatahub-io/opendatahub-operator/v2/apis/dscinitialization/v1"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
+	odhtypes "github.com/opendatahub-io/opendatahub-operator/v2/pkg/controller/types"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/deploy"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/manifest"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/serverless"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/feature/servicemesh"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/metadata/labels"
+)
+
+func kserveManifestInfo(sourcePath string) odhtypes.ManifestInfo {
+	return odhtypes.ManifestInfo{
+		Path:       deploy.DefaultManifestPath,
+		ContextDir: componentName,
+		SourcePath: sourcePath,
+	}
+}
+
+func odhModelControllerManifestInfo(sourcePath string) odhtypes.ManifestInfo {
+	return odhtypes.ManifestInfo{
+		Path:       deploy.DefaultManifestPath,
+		ContextDir: odhModelControllerComponentName,
+		SourcePath: sourcePath,
+	}
+}
+
+func checkDependentOperators(ctx context.Context, cli client.Client) *multierror.Error {
+	var multiErr *multierror.Error
+
+	if found, err := cluster.OperatorExists(ctx, cli, serviceMeshOperator); err != nil {
+		multiErr = multierror.Append(multiErr, err)
+	} else if !found {
+		err = fmt.Errorf("operator %s not found. Please install the operator before enabling %s component",
+			serviceMeshOperator, componentName)
+		multiErr = multierror.Append(multiErr, err)
+	}
+
+	if found, err := cluster.OperatorExists(ctx, cli, serverlessOperator); err != nil {
+		multiErr = multierror.Append(multiErr, err)
+	} else if !found {
+		err = fmt.Errorf("operator %s not found. Please install the operator before enabling %s component",
+			serverlessOperator, componentName)
+		multiErr = multierror.Append(multiErr, err)
+	}
+	return multiErr
+}
+
+func configureServerlessFeatures(dsciSpec *dsciv1.DSCInitializationSpec, kserve *componentsv1.Kserve) feature.FeaturesProvider {
+	return func(registry feature.FeaturesRegistry) error {
+		servingDeployment := feature.Define("serverless-serving-deployment").
```

Comment:
I think the feature tracker here should be owned by the Kserve CR. 
The FT builder has a `OwnedBy` method that should probably be used here ?

Reply:
All of the features will be owned by the Kserve CR, because the `features.FeaturesProvider` returned by this function will be passed to `features.ComponentFeaturesHandler()` (see `configureServerless()` in `kserve_controller_actions.go`). That will result in a `FeaturesHandler` being created, and then when that's all processed, that `OwnedBy` method will be called with the `owner` defined there (which will be the Kserve instance) for all of the Features here.